context.go

```diff
@@ -136,9 +62,11 @@ func (c *Ctx) LoadProject() (*Project, error) {
 
 	// The path may lie within a symlinked directory, resolve the path
 	// before moving forward
-	p.AbsRoot, err = c.resolveProjectRoot(p.AbsRoot)
+	p.AbsRoot, c.GOPATH, err = c.ResolveProjectRootAndGOPATH(p.AbsRoot)
```

Comment:
Rather than having a method, `*Project.ResolvedAbsRoot()`, I think we should be precomputing that information here and storing it in a new property. I don't have a strong feeling on whether that property should be on `Ctx` or `Project`. But, as currently implemented, `*Project.ResolvedAbsRoot()` is effectively discarding errors, which we just can't do. The most convenient solution is to do the work earlier (and just once), here, where there's already an error return path that we can use.

Given this goal, it's probably worth breaking `ResolveProjectRootAndGOPATH()` up into some smaller methods, so that we don't have to add a fourth return value.